src/primitives/transaction.cpp

```diff
@@ -76,9 +76,12 @@ uint256 CTransaction::GetWitnessHash() const
 }
 
 /* For backward compatibility, the hash is initialized to 0. TODO: remove the need for this default constructor entirely. */
-CTransaction::CTransaction() : nVersion(CTransaction::CURRENT_VERSION), vin(), vout(), nLockTime(0), hash() {}
-CTransaction::CTransaction(const CMutableTransaction &tx) : nVersion(tx.nVersion), vin(tx.vin), vout(tx.vout), nLockTime(tx.nLockTime), hash(ComputeHash()) {}
```

Comment:
In commit "Reorder C{,Mutable}Transaction for better packing":

Seems like you could remove ComputeHash now, it doesn't appear to be called anywhere else.